src/topics/expressions.md

```diff
@@ -139,7 +139,67 @@ only in certain fields.  These are:
 
 [file-prop]: https://www.commonwl.org/v1.0/CommandLineTool.html#File
 
+## Using external libraries and inline JavaScript code with `expressionLib`
 
-% TODO
-% - (maybe not before other concepts? move this to after inputs/outputs/etc?)
-% - External libraries and expressionLib - https://github.com/common-workflow-language/user_guide/issues/126
+The requirement `InlineJavascriptRequirement` supports an `expressionLib` attribute
+that allows users to load external JavaScript files, or to provide inline JavaScript
+code.
+
+Entries added to the `expressionLib` attribute are parsed with the JavaScript engine
+of the CWL Runner. This can be used to include external files or to create JavaScript
+functions that can be called in other parts of the CWL document.
+
+```{note}
+
+The `cwltool` CWL Runner, uses a JavaScript Engine that supports
+[ECMAScript 5.1](https://262.ecma-international.org/5.1/). This means that any
+code that you include or write in your CWL Document must be compliant with
+ECMAScript 5.1.
+```
+
+For example, we can write a JavaScript function to an external file `custom-functions.js`:
+
+```{literalinclude} /_includes/cwl/custom-functions.js
+:language: javascript
+:caption: "`custom-functions.js`"
+:name: "`custom-functions.js`"
+```
+
+The following CWL document includes the `custom-functions.js` file and uses its function
+`capitalizeWords`:
+
+```{literalinclude} /_includes/cwl/hello-world-expressionlib.cwl
+:language: cwl
+:caption: "`hello-world-expressionlib.cwl`"
+:name: "`hello-world-expressionlib.cwl`"
+```
+
+```{note}
+In `expressionLib`, `$include` supports both relative and absolute paths.
```

Comment:
@mr-c I added this note as I was curious about it as well, and couldn't find it in the specification. Once this PR is merged, I am planning to update the 1.2.1 branch with some notes about `expressionLib`. Do you know if this (loading relative/absolute paths) is the intended behavior?

If so I will try to word it like that, and also add a note that it doesn't support loading remote files (e.g. `$include: https://cdn.etcetc/underscore.js` is not supported).

Reply:
Remote references should work: https://www.commonwl.org/v1.0/SchemaSalad.html#Include

Reply:
Oh, #TIL! Going to try that too. Do you think we should include an example of that here too?

Reply:
Hmm.. I think a note is enough. Relying on external resources is fragile.